crates/sui-types/src/messages.rs

```diff
@@ -574,7 +582,7 @@ impl<S> TransactionEnvelope<S> {
     /// Get the transaction digest and write it to the cache
     pub fn digest(&self) -> &TransactionDigest {
         self.transaction_digest
-            .get_or_init(|| TransactionDigest::new(sha3_hash(&self.data)))
+            .get_or_init(|| TransactionDigest::new(sha3_hash(&self.signed_data)))
```

Comment:
Can we verify with wallet devs that they also know how to compute the `transaction_digest` now that it's computed differently? Mostly to ensure backwards compatibility @666lcz @pchrysochoidis @gegaowp  (FYI previously it didn't include user's sig, now the sig is prepended to the data we used to hash to derive the tx_digest)

Reply:
Checked out with @666lcz, and looks like wallet, sdk, nor explorer are computing txIds. Merging!